packages/theme-addresses/__tests__/theme-addresses.test.js

```diff
@@ -0,0 +1,231 @@
+  });
+
+  test('populates the given country and province selectors when a default is set for country', () => {
+    document.body.innerHTML = `<form>
+      <select id="country" data-default="New Zealand"></select>
+      <select id="province"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#country');
+    const provinceSelector = document.querySelector('#province');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector);
+
+    expect(countrySelector.value).toEqual('New Zealand');
+    expect(provinceSelector.querySelectorAll('option').length).not.toEqual(0);
+    expect(provinceSelector.value).toEqual('Auckland');
+    expect(provinceSelector.querySelectorAll('option[value="Auckland"]').length).toEqual(1);
+  });
+
+  test('selects the first province in list if default value is not found as part of available options', () => {
+    document.body.innerHTML = `<form>
+      <select id="country" data-default="Canada"></select>
+      <select id="province" data-default="Auckland"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#country');
+    const provinceSelector = document.querySelector('#province');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector);
+
+    expect(countrySelector.value).toEqual('Canada');
+    expect(provinceSelector.querySelectorAll('option').length).not.toEqual(0);
+    expect(provinceSelector.value).toEqual('Alberta');
+  });
+
+  test('populates the given country and province selectors when a default is set for country and province', () => {
+    document.body.innerHTML = `<form>
+      <select id="country" data-default="Canada"></select>
+      <select id="province" data-default="Quebec"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#country');
+    const provinceSelector = document.querySelector('#province');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector);
+
+    expect(countrySelector.value).toEqual('Canada');
+    expect(provinceSelector.value).toEqual('Quebec');
+  });
+
+  test('triggers onCountryChange callback', (done) => {
+    document.body.innerHTML = `<form>
+      <select id="country" data-default="Canada"></select>
+      <select id="province"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#country');
+    const provinceSelector = document.querySelector('#province');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector, {
+      onCountryChange: (provinces, provinceNodeElement, countryNodeElement) => {
```

Comment:
A more precise test might look like:
```
const mockFunction = jest.fn();
countryProvinceSelector.build(countrySelector, provinceSelector, {
      onCountryChange: mockFunction
});

countrySelector.dispatchEvent(new Event('change'));

expect(mockFunction).toHaveBeenCalled();

Reply:
oh yea .. I forgot about the mock functions